backend/src/packages/chats/chat.service.ts

```diff
@@ -0,0 +1,52 @@
+import { type Service } from '#libs/types/types.js';
+
+import { type ChatRepository } from './chat.repository.js';
+import {
+  type ChatGetAllItemResponseDto,
+  type ChatGetAllResponseDto,
+  type CreateChatPayload,
+} from './libs/types/types.js';
+
+class ChatService implements Service {
+  private chatRepository: ChatRepository;
+
+  public constructor(chatRepository: ChatRepository) {
+    this.chatRepository = chatRepository;
+  }
+
+  public find(): ReturnType<Service['find']> {
+    return Promise.resolve(null);
+  }
+
+  public async findAll(): ReturnType<Service['findAll']> {
+    return await Promise.resolve({ items: [] });
+  }
+
+  public async findAllByUserId(userId: number): Promise<ChatGetAllResponseDto> {
+    const items = await this.chatRepository.findAllByUserId(userId);
+
+    return {
+      items: items.map((item) => {
+        return item.toObject();
+      }),
+    };
+  }
+
+  public async create(
+    payload: CreateChatPayload,
+  ): Promise<ChatGetAllItemResponseDto> {
+    const item = await this.chatRepository.create(payload);
```

Comment:
don't pass payload directly, use initializeNew
```suggestion
    const item = await this.chatRepository.create(payload);
```

Reply:
If so, I would need to type members as `UserToChatQueryResponse[] | number[]`, or create a `users-to-chats.entity` to initialize it first and then pass it to the chat entity. What do you think will be the preferable way?

Reply:
@Liza-Veis, as we decided to remove members from the new initializer, do I still need to create here a chat entity, and pass it to the create method with members argument separately?

Reply:
yes, use initializeNew to pass basic entry and pass members as a second parameter